src/Proxy/ContainerInterfaceProxy.php

```diff
@@ -0,0 +1,260 @@
+            ContainerInterface::class,
+            get_class($this->container),
+            $method,
+            $arguments,
+            $result,
+            $this->getCurrentResultStatus(),
+            $error,
+            $timeStart,
+            microtime(true),
+            );
+    }
+
+    private function logToEvent(string $method, ?array $arguments, $result, ?object $error, float $timeStart): void
+    {
+        $this->config->getDispatcher()->dispatch(new ProxyMethodCallEvent(
+            ContainerInterface::class,
+            get_class($this->container),
+            $method,
+            $arguments,
+            $result,
+            $this->getCurrentResultStatus(),
+            $error,
+            $timeStart,
+            microtime(true),
+            ));
+    }
+
+    private function isDecorated(string $service): bool
+    {
+        return isset($this->config->getDecoratedServices()[$service]) || in_array($service, $this->config->getDecoratedServices(), true);
+    }
+
+    private function getServiceProxy(string $service, object $instance): ?object
+    {
+        if (!$this->isDecorated($service)) {
+            return null;
+        }
+
+        if (isset($this->config->getDecoratedServices()[$service]) && is_callable($this->config->getDecoratedServices()[$service])) {
+            return $this->getServiceProxyFromCallable($service, $instance);
+        } elseif (isset($this->config->getDecoratedServices()[$service]) && is_array($this->config->getDecoratedServices()[$service]) &&
+            !isset($this->config->getDecoratedServices()[$service][0])) {
+            return $this->getCommonMethodProxy($service, $instance, $this->config->getDecoratedServices()[$service]);
+        } elseif (isset($this->config->getDecoratedServices()[$service]) && is_array($this->config->getDecoratedServices()[$service])) {
+            return $this->getServiceProxyFromArray($service, $instance);
+        } elseif (interface_exists($service) && ($this->config->getCollector() !== null || $this->config->getDispatcher() !== null)) {
+            return $this->getCommonServiceProxy($service, $instance);
+        }
+
+        return null;
+    }
+
+    private function getCommonMethodProxy(string $service, object $instance, array $callbacks): ?object
+    {
+        $methods = [];
+        while ($callback = current($callbacks)) {
+            $method = key($callbacks);
+            next($callbacks);
+            if (is_string($method) && is_callable($callback)) {
+                $methods[$method] = $callback;
+            }
+        }
+
+        return $this->proxyManager->createObjectProxyFromInterface(
+            $service,
+            CommonMethodProxy::class,
+            [$service, $instance, $methods, $this->config->getCollector(), $this->config->getDispatcher(), $this->config->getLogLevel()]
+        );
+    }
+
+    private function getServiceProxyFromCallable(string $service, object $instance): ?object
```

Comment:
not used $instance

Reply:
I agree,  it is temporary, may be removed.

Reply:
Done